### ISSUE TITLE
`clippy::missing_docs_in_private_items` detects `const` in `fn` but `--document-private-items` doesn't generate it.

### ISSUE THREAD

**Issue by nogiro** [rust-lang/rust-clippy#13298]
### Summary

`clippy::missing_docs_in_private_items` seems to trigger in `const` in `fn`, but `cargo doc --document-private-items` doesn't generate this `const`.

### Lint Name
missing_docs_in_private_items

### Reproducer

I tried this code:

```rust
#![warn(clippy::missing_docs_in_private_items)]

//! `const` in `fn`.

/// The application entry point.
fn main() {
    const MSG: &str = "Hello, world!";
    println!("{MSG}");
}
```

I expected to see this happen:

```
warning: missing documentation for a constant
 --> src/main.rs:7:5
  |
7 |     const MSG: &str = "Hello, world!";
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_docs_in_private_items
note: the lint level is defined here
 --> src/main.rs:1:9
  |
1 | #![warn(clippy::missing_docs_in_private_items)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: `missing_docs_in_private_items_detects_const_in_fn` (bin "missing_docs_in_private_items_detects_const_in_fn") generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.00s
```

Instead, this happened:

No warning.

### Version

```text
rustc 1.80.1 (3f5fd8dd4 2024-08-06)
binary: rustc
commit-hash: 3f5fd8dd41153bc5fdca9427e9e05be2c767ba23
commit-date: 2024-08-06
host: x86_64-unknown-linux-gnu
release: 1.80.1
LLVM version: 18.1.7
```
